src/demikernel/bindings.rs

```diff
@@ -48,14 +48,14 @@ pub extern "C" fn demi_init(args: *const demi_args_t) -> c_int {
         args.log_callback
     };
 
-    if log_callback.is_none() {
-        logging::initialize();
-    } else {
+    if let Some(log_callback) = log_callback {
```

Comment:
do we prefer a match statement there or the if let?

Reply:
The Rust ebook says that `if let ... else` is a more concise way to represent this logic (especially when we care about just 2 patterns which is exhaustive in this case).

https://doc.rust-lang.org/stable/book/ch06-03-if-let.html

Match would result in a slightly more verbose representation and is generally good for exhaustive pattern match for more than 2 obvious patterns, but has the same possible patterns for a boolean like in this case. I don't have strong opinion about this, as far as we are consistent and the code flows easily for the reader.